piecrust-uplink/src/abi/state.rs

```diff
@@ -292,6 +294,37 @@ pub fn owner<const N: usize>(contract: ContractId) -> Option<[u8; N]> {
     }
 }
 
+/// Returns given contract's free limit, if the contract exists and if it
+/// has a free limit set.
+pub fn free_limit(contract: ContractId) -> Option<u64> {
+    let contract_id_ptr = contract.as_bytes().as_ptr();
+    unsafe {
+        match ext::free_limit(contract_id_ptr) as usize {
+            0 => None,
+            arg_pos => with_arg_buf(|buf, _| {
+                let ret = archived_root::<Option<u64>>(&buf[..arg_pos]);
+                ret.deserialize(&mut Infallible).expect("Infallible")
+            }),
+        }
+    }
+}
+
+/// Returns given contract's free gas price hint, if the contract exists and
+/// if it has a free price hint set.
+pub fn free_price_hint(contract: ContractId) -> Option<(u64, u64)> {
+    let contract_id_ptr = contract.as_bytes().as_ptr();
+
+    unsafe {
+        match ext::free_price_hint(contract_id_ptr) as usize {
+            0 => None,
+            arg_pos => with_arg_buf(|buf, _| {
+                let ret = archived_root::<Option<(u64, u64)>>(&buf[..arg_pos]);
+                ret.deserialize(&mut Infallible).expect("Infallible")
```

Comment:
The same applies here, but I can see it being more complicated due to it being a tuple.

Reply:
sounds good

Reply:
done